Source/WebCore/page/DOMSelection.cpp

```diff
@@ -416,6 +433,40 @@ ExceptionOr<void> DOMSelection::removeRange(Range& liveRange)
     return { };
 }
 
+Vector<Ref<StaticRange>> DOMSelection::getComposedRanges(FixedVector<std::reference_wrapper<ShadowRoot>>&& shadowRoots)
+{
+    auto frame = this->frame();
+    if (!frame)
+        return { };
+    auto range = frame->selection().selection().range();
+    if (!range)
+        return { };
+
+    HashSet<Ref<ShadowRoot>> shadowRootSet;
```

Comment:
Any benefit to reserving initial capacity here?

Reply:
Ah, good point. Will do.